File: Dart/Building_Core_Of_Flutter/Flutter_Internal_01/16Jan/Program4.dart
class Demo{
  Demo(){
    print("In Demo Constructor");
  }
  void fun(){
    print("In Fun method");
  }
}
Demo objFun(){
  print("In objFun function");
  return Demo();
}
void main(){
  Demo obj = objFun();
  obj.fun();
}

File: Dart/Essentials_of_Dart/Singleton_Design_Pattern/31Oct/Program4.java
class Demo{
    static Demo obj = new Demo();

    private Demo(){
        System.out.println("In Constructor");
    }

    static Demo getInstance(){
        return obj;
    }
}

class Client{
    public static void main(String args[]){
        Demo obj1 = Demo.getInstance();
        Demo obj2 = Demo.getInstance();
        Demo obj3 = Demo.getInstance();
    }
}